packages/frontend/src/context/User.ts

```diff
@@ -340,32 +340,32 @@ export class User {
         if (this.reputation >= 0) throw new Error('do not need to airdrop')
 
         await this.unirepConfig.loadingPromise
-        const unirepSocial = new ethers.Contract(
-            this.unirepConfig.unirepSocialAddress,
-            config.UNIREP_SOCIAL_ABI,
-            config.DEFAULT_ETH_PROVIDER
-        )
+
         // generate an airdrop proof
         const negRep = Math.min(
             Math.abs(this.reputation),
-            this.unirepConfig.defaultAirdropKarma
+            this.unirepConfig.subsidy
         )
 
         const negRepProof = await this.userState.genNegativeRepProof(
             BigInt(this.unirepConfig.attesterId),
             BigInt(negRep)
         )
 
-        const epk = genEpochKey(
+        // Check if the user already got airdropped
+        const { number: currentEpoch } =
+            await this.userState?.loadCurrentEpoch()
+        const subsidyKey = genEpochKey(
             this.id.identityNullifier,
-            await this.userState.getUnirepStateCurrentEpoch(),
-            0
+            currentEpoch,
+            0,
+            this.unirepConfig.epochTreeDepth
         )
-        // Check if the user already got airdropped
-        const subsidy = (
-            await this.unirepConfig.unirepSocial.subsidy()
-        ).toNumber()
-        if (subsidy !== this.unirepConfig.defaultAirdropKarma) {
+        const spentSubsidy = await this.unirepConfig.unirepSocial.subsidies(
+            currentEpoch,
+            subsidyKey
+        )
+        if (spentSubsidy >= this.unirepConfig.subsidy) {
```

Comment:
`this.unirepConfig.subsidy` is always `30`
`spentSubsidy` should be `0` to `30`
so here it should check `if(spentSubsidy.toNumber() > 0)`